greens/tasks.py

```diff
@@ -54,3 +68,46 @@ def allocate_leave():
 					'new_leaves_allocated': flt(earned_leaves)
 				}).submit()
 	return
+
+	def half_day(doc, method=None):
+	shift_checkout(doc)
+		if doc.status == 'Half Day':
+			logs = frappe.db.get_list('Employee Checkin', fields="*", filters={
+				'skip_auto_attendance':'0',
+				'employee': ['=',doc.employee],
+				'time': ['>',today()],
+				'time': ['<', add_to_date(today(), days=1, as_string=True)],
+				}, order_by="employee,time")
+				total_working_hours = calculate_working_hours(
+				logs,
+				'Strictly based on Log Type in Employee Checkin',
+				'First Check-in and Last Check-out'
+			)[0]
+			if int(total_working_hours) < 5:
+				frappe.throw('Not completed 5 Hours')
+
+	def shift_checkout(doc):
+		emp_details = frappe.get_all('Employee Checkin',filters={
+			'employee': ['=',doc.employee],
+			'time': ['>',today()],
+			'time': ['<', add_to_date(today(), days=1, as_string=True)],
+		},
+		fields=['employee_name', 'employee','count(name) as count','log_type','time'],
+		group_by="log_type",
+		)
+		total_in = 0
+		total_out = 0
+		for row in emp_details:
```

Comment:
What is the purpose of this variable and this loop? Shouldn't `count` give the required data already?

Reply:
remove loop and update code